src/dex-ui/hooks/hederaTokenService/useFetchTokenData.tsx

```diff
@@ -0,0 +1,28 @@
+import { HTSMutations } from "./types";
+import { useMutation } from "react-query";
+import { DexService, MirrorNodeTokenById } from "../../services";
+import { isNil } from "ramda";
+
+export interface UseFetchTokenDataParams {
+  tokenId: string;
+}
+
+export function useFetchTokenData(
+  handleTokenSuccessResponse: (tokenData: MirrorNodeTokenById) => void,
+  handleTokenErrorResponse: () => void
+) {
+  return useMutation<MirrorNodeTokenById, Error | undefined, UseFetchTokenDataParams, HTSMutations.TokenData>(
```

Comment:
Mutations should be used for create/update/delete actions. We should use `useQuery` for fetching data. You should be able to use `refetch` if you need to perform subsequent query calls after the page renders.

User updates token id input => make sure token id value that will be passed to `useFetchTokenData` is updated => call `refetch`

Reply:
Done